2-projects/1-todo-list/script.js

```diff
@@ -1,6 +1,45 @@
 function populateTodoList(todos) {
-  let list = document.getElementById("todo-list");
+  let ul = document.getElementById("todo-list");
   // Write your code to create todo list elements with completed and delete buttons here, all todos should display inside the "todo-list" element.
+  for (let [index, item] of todos.entries()) {
+    // wtf is this?
+    let li = document.createElement("li");
+    li.innerText = item.task;
+    li.classList.add(
+      "list-group-item",
+      "d-flex",
+      "justify-content-between",
+      "align-items-center"
+    );
+    let span = document.createElement("span"),
+      icon1 = document.createElement("i"),
+      icon2 = document.createElement("i");
+    span.classList.add("badge", "bg-primary", "rounded-pill");
+    icon1.classList.add("fa", "fa-check");
+    icon1.style.marginRight = "0.25rem";
+    icon2.classList.add("fa", "fa-trash");
+    icon1.ariaHidden = "true";
+    icon2.ariaHidden = "true";
+    span.append(icon1, icon2);
+    li.append(span);
+    ul.append(li);
+    icon1.addEventListener("click", () => {
+      if (
+        icon1.parentElement.parentElement.style.textDecoration ===
+        "line-through"
+      ) {
+        icon1.parentElement.parentElement.style.textDecoration = "none";
+      } else {
+        icon1.parentElement.parentElement.style.textDecoration = "line-through";
+      }
+    });
```

Comment:
The code is quite complex and hard to follow, I'd suggest breaking it down into several distinct functions.